### PR TITLE
Display deck names when deleting

### PR DESCRIPTION
**Pull request by escribapetrus**
This change displays the name of the deck deleted from the browser or the sidebar.
Closes [#3047](https://github.com/ankitects/anki/issues/3047)

- update python files to get deck name from frontend state.
- add message template to ftl

Screenshot from 2024-03-07 12-34-45

### CODE REVIEW

**Comment by dae** [comment]
Thanks for looking into this! While your implementation looks solid, having the deck removal method return a list of deck names feels a bit "unclean", as it makes the method have multiple responsibilities, and increases the number of opchanges that code needs to handle. It's also going to cause an unreadably-large tooltip if the user happens to delete a lot of decks at once.

My assumption was that users were worried about accidentally deleting a deck from the deck list screen, where only a single deck would be deleted at once (and the deck name can be extracted from the state in deckbrowser.py). I may be wrong about that assumption, as users can also delete decks from the browse screen, though again that's only one deck at once, unless the user has enabled multi-selection mode.

How would you feel about reverting the changes to RemoveDecks, and instead getting the deck name from the existing state (like  _collapse() in deckbrowser and the state in the sidebar)?

**Comment by escribapetrus** [comment]
@dae Thank you. Will do this in the weekend, and while at it I guess I'll get a better understanding of the system.

While I have you here, could you tell me how to access the database, or at least where it's located? I'd like to connect to it on the SQL client to get a better understanding of the architecture. Sorry for bothering.

**Comment by dae** [comment]
It's an SQLite file, and can be accessed with sqlitebrowser or the sqlite command line tool.

https://docs.ankiweb.net/files.html#file-locations

File: pylib/anki/decks.py [hunk 1]
```diff
@@ -33,7 +33,7 @@
 
 DeckId = NewType("DeckId", int)
 DeckConfigId = NewType("DeckConfigId", int)
-
+DeckName = NewType("DeckName", str)
```
[the remark this answers is not in the document]
Comment (escribapetrus) [reply pylib/anki/decks.py:36]: > Nearly there. Were the two empty line removals deliberate?

Not really, I'll fix that.

> A newtype for this feels a bit unnecessary - I think a regular string would suffice.

Okay, I'll change that in a second.

**Comment by dae** [comment]
Looks good - thanks Pedro!

**Pull request merged**